src/domains/shielder/utils/useActivityModal.tsx

```diff
@@ -0,0 +1,49 @@
+import { useCallback } from 'react';
+
+import { useModals } from 'src/domains/misc/components/Modal';
+import ActivityDetailsModal from 'src/domains/shielder/components/TransactionDetailsModal';
+import { PartialLocalShielderActivityHistory } from 'src/domains/shielder/stores/getShielderIndexedDB';
+import { useActivityHistory } from 'src/domains/shielder/utils/useActivityHistory';
+
+/**
+ * Custom hook to handle transaction details modal logic.
+ * Fetches the latest transaction data and opens a modal with up-to-date information.
+ * Handles modal ID transitions when transactions move from pending (localId only)
+ * to confirmed state (both localId and txHash), preventing duplicate modals.
+ */
+const useActivityModal = () => {
+  const { mount, updateId } = useModals();
+  const { refetch } = useActivityHistory();
+
+  const openTransactionModal = useCallback(async (tx: PartialLocalShielderActivityHistory) => {
+    const { data } = await refetch();
+    const thisTransaction =
+      data?.find(transaction => transaction.txHash === tx.txHash || transaction.localId === tx.localId);
+
+    const modalId = tx.txHash ?? tx.localId;
```

Comment:
I'd really want to get rid of `updateId` logic. Would using `tx.localId` and ignoring the `tx.txHash` allow us to remove the `updateId` logic and fix the duplicated modal issue?